src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+                    if ( evidence2.getContigStart() >= reportableEnd ) break;
+                    reportableEvidence.add(evidence2);
+                    itr.remove();
+                }
+            }
+            recordSet.add(evidence);
+            return reportableEvidence.iterator();
+        }
+    }
+
+    /**
+     * Minimalistic simple interval.
+     */
+    @DefaultSerializer(Interval.Serializer.class)
+    @VisibleForTesting static final class Interval {
+        private final int contig;
+        private final int start;
+        private final int end;
+
+        Interval( final int contig, final int start, final int end ) {
+            this.contig = contig;
+            this.start = start;
+            this.end = end;
+        }
+
+        private Interval( final Kryo kryo, final Input input ) {
+            contig = input.readInt();
+            start = input.readInt();
+            end = input.readInt();
+        }
+
+        private void serialize( final Kryo kryo, final Output output ) {
+            output.writeInt(contig);
+            output.writeInt(start);
+            output.writeInt(end);
+        }
+
+        public int getContig() { return contig; }
+        public int getStart() { return start; }
+        public int getEnd() { return end; }
+        public int getLength() { return end-start; }
+
+        public boolean isDisjointFrom( final Interval that ) {
+            return this.contig != that.contig || this.end < that.start || that.end < this.start;
+        }
+
+        public Interval join( final Interval that ) {
+            if ( this.contig != that.contig ) throw new GATKException("Joining across contigs.");
+            return new Interval(contig, Math.min(this.start, that.start), Math.max(this.end, that.end));
+        }
+
+        @Override
+        public boolean equals( final Object obj ) {
+            if ( !(obj instanceof Interval) ) return false;
+            final Interval that = (Interval)obj;
+            return this.contig == that.contig && this.start == that.start && this.end == that.end;
+        }
+
+        @Override
+        public int hashCode() {
+            return 47*(47*(47*(47*contig)+start)+end);
+        }
+
+        public static final class Serializer extends com.esotericsoftware.kryo.Serializer<Interval> {
+            @Override
+            public void write( final Kryo kryo, final Output output, final Interval interval ) {
+                interval.serialize(kryo, output);
+            }
+
+            @Override
+            public Interval read( final Kryo kryo, final Input input, final Class<Interval> klass ) {
+                return new Interval(kryo, input);
+            }
+        }
+    }
+
+    /**
+     * A class to examine a stream of BreakpointEvidence, and group it into Intervals.
+     */
+    private static final class IntervalMapper implements Function<BreakpointEvidence, Iterator<Interval>> {
```

Comment:
The name IntervalMapper doesn't really capture what this class is doing, for me at least. Can't think of a much better name off the top of my head but even `EvidenceToIntervalMapper` would do more for me.